cloudfour/pwa-workshop
# Feature: Exercise 6, Manifest + Add to Home Screen/Install

## Overview

This PR adds a `manifest.json` plus "Add to Home Screen" logic.

## Testing

1. In Chrome, load up https://deploy-preview-27--festive-keller-9a29cb.netlify.com and confirm:
    - you see the "Install PWA Workshop App" button

    <img width="965" alt="Screen Shot 2019-06-07 at 12 06 05 PM" src="https://user-images.githubusercontent.com/459757/59079002-b62cf500-891c-11e9-93d8-dd21bf5c5f89.png">


1. Click the "Install PWA Workshop App" button and confirm you see the "Install App?" dialog

    ![Screen Shot 2019-06-07 at 11 35 56 AM](https://user-images.githubusercontent.com/459757/59078522-795ffe80-891a-11e9-8e95-7b0aaeef902b.png)

1. Choose "Cancel" and confirm the "Install PWA Workshop App" is enabled

1. Click the "Install PWA Workshop App" again but this time choose "Install" and confirm
    - the PWA installs and opens
    - the "Install PWA Workshop App" is not visible
    ![Screen Shot 2019-06-07 at 11 46 14 AM](https://user-images.githubusercontent.com/459757/59078566-b88e4f80-891a-11e9-8fb6-577ab8d856fe.png)

1. Load up https://deploy-preview-27--festive-keller-9a29cb.netlify.com in a browser and confirm:
    - the "Install PWA Workshop App" button is not visible (because the app is already installed)

1. Uninstall the app
    ![delete-app](https://user-images.githubusercontent.com/459757/59078638-fdb28180-891a-11e9-9223-ed7e71cf4776.gif)

1. Load up https://deploy-preview-27--festive-keller-9a29cb.netlify.com in a browser and confirm:
    - the "Install PWA Workshop App" button is visible again (because the app is no longer installed)

---

- [Trello Card](https://trello.com/c/9zE1BUXl/48-service-worker-exercise-6-code)
